charts/seed-bootstrap/charts/vpa/charts/runtime/templates/deployment-admission-controller.yaml

```diff
@@ -47,7 +47,10 @@ spec:
         {{- if .Values.admissionController.registerByURL }}
         - --register-by-url=true
         - --webhook-address=https://vpa-webhook.{{.Release.Namespace}}
-        - --webhook-port={{ .Values.admissionController.port }}
+        - --webhook-port={{ .Values.admissionController.servicePort }} # the port is only respected if register-by-url is true, that's why it's in this if-block
+                                                                       # if it's false it will not set the port during registration, i.e., it will be defaulted to 443,
+                                                                       # so the servicePort has to be 443 in this case
+                                                                       # see https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/config.go#L70-L74
```

Comment:
I don't understand the reasoning behind this change. There is `{{- if .Values.admissionController.registerByURL }}`, so the important thing is what happens if `register-by-url` is true. In this case, the Webhook URL is passed directly to the API server (there is no service), and therefore the port above must be the port on which the webhook server listens to, that is 10250 by default, not 443. What am I missing?

Reply:
You can check the comment here: https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/main.go#L67

The self-registration process is constructing the final URL out of the values passed by `--webhook-address` and `--webhook-port`, see https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/main.go#L133.

Earlier, the `--webhook-port` was set to the server port (10250) which was wrong - now it's the service port (443)

Reply:
OK, I got it, so here if `.Values.admissionController.registerByURL` we are still passing the *service* address and port. Why do we ever need to do this? I thought this option would be useful only if there is no service or for some reason we don't want to use it?

Reply:
I answered my own question, never mind.